sbomtool/cmd/sbomtool/main.go

```diff
@@ -0,0 +1,184 @@
+	fmt.Println("  --cyclonedx          Generate a CycloneDX SBOM")
+	fmt.Println("\nMerge Options:")
+	fmt.Println("  --level int          Merge level (default 0)")
+	fmt.Println("\nExamples:")
+	fmt.Println("  sbomtool --log-level debug generate --name myapp --build-dir ./build --out-dir ./sbom --spdx")
+	fmt.Println("  sbomtool generate --name myapp --build-dir ./build --out-dir ./sbom --cyclonedx")
+}
+
+// findSubcommandPos finds the position of the first non-flag argument (the subcommand)
+func findSubcommandPos() int {
+	for i := 1; i < len(os.Args); i++ {
+		if !strings.HasPrefix(os.Args[i], "-") {
+			return i
+		}
+	}
+	return len(os.Args)
+}
+
+// run executes the main application logic and returns an exit code.
+// This separation allows for better testability by avoiding direct os.Exit calls.
+func run() int {
+	// Define global flags
+	globalFlags := flag.NewFlagSet("sbomtool", flag.ExitOnError)
+	helpFlag := globalFlags.Bool("help", false, "Show help message")
+	logLevelFlag := globalFlags.String("log-level", "info", "Log level (debug, info, warn, error)")
+
+	// Find the position of the first non-flag argument (the subcommand)
+	subcommandPos := findSubcommandPos()
+
+	// If there are no arguments or only flags, parse them
+	if subcommandPos == len(os.Args) {
+		globalFlags.Parse(os.Args[1:])
+		// Show help if no subcommand is provided or help flag is set
+		if *helpFlag || subcommandPos == len(os.Args) {
+			printHelp()
+			return 0
+		}
+	} else {
+		// Parse only the global flags before the subcommand
+		globalFlags.Parse(os.Args[1:subcommandPos])
+	}
+
+	// Configure logging based on global flag
+	logging.Configure(logging.LogLevel(*logLevelFlag))
+
+	// Show help if help flag is set
+	if *helpFlag {
+		printHelp()
+		return 0
+	}
+
+	// If no subcommand is provided or "help" is the subcommand
+	if subcommandPos == len(os.Args) || os.Args[subcommandPos] == "help" {
+		printHelp()
+		return 0
+	}
+
+	// Get the subcommand
+	subcommand := os.Args[subcommandPos]
+
+	switch subcommand {
+	case "generate":
+		generateCmd := flag.NewFlagSet("generate", flag.ExitOnError)
+		generateName := generateCmd.String("name", "", "Name of the target package")
+		generateBuilDir := generateCmd.String("build-dir", "", "Target directory of the package")
+		generateOutDir := generateCmd.String("out-dir", "", "Output directory for the SBOM")
+		generateSpdx := generateCmd.Bool("spdx", false, "Generate an SPDX SBOM")
+		generateCyclonedx := generateCmd.Bool("cyclonedx", false, "Generate a CycloneDX SBOM")
+
+		// Parse the remaining arguments
+		if err := generateCmd.Parse(os.Args[subcommandPos+1:]); err != nil {
+			fmt.Fprintf(os.Stderr, "Error parsing generate command: %v\n", err)
+			return 1
+		}
+
+		slog.Info("Starting sbomtool generate",
+			"name", *generateName,
+			"build_dir", *generateBuilDir,
+			"out_dir", *generateOutDir,
+			"spdx", *generateSpdx,
+			"cyclonedx", *generateCyclonedx,
```

Comment:
Shouldn't this be `debug` given all the information that was passed? Also, another place to use a widely used logging library `logrus`. This is used by projects like containerd. The advantage being that you can provide `context` to the logs.

Reply:
`logrus` predates `slog` but `slog` is now in the standard library, so waiting on golang experts to weigh in, it seems like a new project should prefer it.

This could definitely work as `debug`